deploy/admission_webhook.yaml

```diff
@@ -0,0 +1,104 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: service-apis
+  labels:
+    name: service-apis
+---
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingWebhookConfiguration
+metadata:
+  name: service-apis-admission
+webhooks:
+  - name: validate.networking.x-k8s.io
+    matchPolicy: Equivalent
+    rules:
+      - operations: [ "CREATE" , "UPDATE" ]
+        apiGroups: [ "networking.x-k8s.io" ]
+        apiVersions: [ "v1alpha1" ]
+        resources: [ "httproutes" ]
+    failurePolicy: Fail
+    sideEffects: None
+    admissionReviewVersions:
+      - v1
+    clientConfig:
+      service:
+        name: service-apis-admission
+        namespace: service-apis
+        path: "/validate"
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    name: service-apis-webhook
+    version: 0.0.1
+  name: service-apis-admission
+  namespace: service-apis
+spec:
+  type: ClusterIP
+  ports:
+    - name: https-webhook
+      port: 443
+      targetPort: 8443
+  selector:
+    name: service-apis-admission
+---
+# TODO(cmluciano): Tone down logging level for webhook
+# once we are satified with it's stability.
+#
+# TODO(cmluciano): Swap image name to the k8s official
+# namespace once kubernetes/k8s.io#1514 is merged.
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: service-apis-admission
+  namespace: service-apis
+  labels:
+    name: service-apis-admission
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      name: service-apis-admission
+  template:
+    metadata:
+      name: service-apis-admission
+      labels:
+        name: service-apis-admission
+    spec:
+      containers:
+        - name: webhook
+          image: cmluciano/service-apis-admission:latest
+          imagePullPolicy: Always
+          args:
+            - -alsologtostderr
+            - --log_dir=/tmp
+            - --tlsCertFile=/etc/certs/cert
+            - --tlsKeyFile=/etc/certs/key
+            - -v=10
+            - 2>&1
+          ports:
+            - containerPort: 8443
+              name: webhook
+          resources:
+            limits:
+              memory: 50Mi
+              cpu: 300m
+            requests:
+              memory: 50Mi
+              cpu: 300m
+          volumeMounts:
+            - name: webhook-certs
+              mountPath: /etc/certs
+              readOnly: true
+            - name: logs
+              mountPath: /tmp
+          securityContext:
+            readOnlyRootFilesystem: true
```

Comment:
How does this work with the log_dir specified above? Is it possible to just log to stdout/stderr instead?

Reply:
removed this bit